File: backend/server-node/src/db/schemas/progress_log.js
const { EntitySchema } = require('typeorm');

const ProgressLogSchema = new EntitySchema({
    name: 'ProgressLog',
    tableName: 'progress_log',
    columns: {
        id: {
            primary: true,
            type: 'int',
            generated: true
        },
        user_id: {
            type: 'int',
            nullable: false
        },
        started_at: {
            type: 'timestamp',
            nullable: false
        },
        type: {
            type: 'enum',
            enum: ['video_call', 'solo_learning'],
            nullable: false
        },
        data: {
            type: 'jsonb',
            nullable: true
        },
        created_at: {
            type: 'timestamp',
            createDate: true,
            nullable: false
        },
        updated_at: {
            type: 'timestamp',
            updateDate: true,
            nullable: false
        }
    },
    relations: {
        user: {
            target: 'User',
            type: 'many-to-one',
            joinColumn: {
                name: 'user_id',
                referencedColumnName: 'id'
            },
            nullable: false
        }
    }
});

module.exports = ProgressLogSchema;

File: backend/server-py/app/src/db/schemas/progress_log.py
from .base import Base, TimestampMixin
import sqlalchemy as sa
from sqlalchemy.dialects import postgresql as pg

class ProgressLog(Base, TimestampMixin):
    __tablename__ = 'progress_log'
    id = sa.Column(sa.Integer, primary_key=True)
    user_id = sa.Column(sa.Integer, sa.ForeignKey('users.id'), nullable=False)
    started_at = sa.Column(sa.DateTime, nullable=False)
    type = sa.Column(sa.Enum('video_call', 'solo_learning', name='learning_type'), nullable=False)
    data = sa.Column(pg.JSONB, nullable=True)  # 학습 데이터
    # data 구조:
    # {
    #   "목표": ["깡총", "차근"],
    #   "week": 1,
    #   "step": 1,
    #   "data": {
    #     "성공_여부": true,
    #     "progress": 0.8
    #   }
    # }
